packages/nexrender-core/src/tasks/download.js

```diff
@@ -76,6 +70,16 @@ const download = (job, settings, asset) => {
             return fetch(src, asset.params || {})
                 .then(res => res.ok ? res : Promise.reject({reason: 'Initial error downloading file', meta: {src, error: res.error}}))
                 .then(res => {
+                    // Set a file extension based on content-type header if not already set
+                    if (path.extname(path.basename(asset.dest)) === '') {
```

Comment:
* this if should rather be `if (!asset.extension)`
* if extension is found you should also update the `asset.extension`, to replicate the data2buf behavior